jupyterdrive/mixednbmanager.py

```diff
@@ -4,68 +4,138 @@
 # Distributed under the terms of the Modified BSD License.
 
 from IPython.html.services.contents.manager import ContentsManager
-from IPython.html.services.contents.filemanager import FileContentsManager
-from .clientsidenbmanager import ClientSideContentsManager
+from IPython.utils.traitlets import List
+from IPython.utils.importstring import import_item
 
 class MixedContentsManager(ContentsManager):
-    DRIVE_PATH_SENTINEL = 'gdrive'
+
+    filesystem_scheme = List([
+            {
+                'root':'local',
+                'contents':"IPython.html.services.contents.filemanager.FileContentsManager"
+            },
+            {
+                'root': 'gdrive',
+                'contents': 'jupyterdrive.clientsidenbmanager.ClientSideContentsManager'
+            }
+        ],
+    help="""
+    List of virtual mount point name and corresponding contents manager
+    """, config=True)
 
     def __init__(self, **kwargs):
-        self.file_contents_manager = FileContentsManager()
-        self.client_side_contents_manager = ClientSideContentsManager()
 
-    def is_drive_path(self, path):
-        components = path.split('/');
-        return components and components[0] == self.DRIVE_PATH_SENTINEL
+        super(MixedContentsManager, self).__init__(**kwargs)
+        self.managers = {}
+
+        ## check consistency of scheme.
+        if not len(set(map(lambda x:x['root'], self.filesystem_scheme))) == len(self.filesystem_scheme):
+            raise ValueError('Scheme should not mount two contents manager on the same mountpoint')
+
+        kwargs.update({'parent':self})
+        for scheme in self.filesystem_scheme:
+            manager_class = import_item(scheme['contents'])
+            self.managers[scheme['root']] = manager_class(**kwargs)
+
+
+    def path_dispatch1(method):
+        def _wrapper_method(self, path, *args, **kwargs):
+            path = path.strip('/')
+            (sentinel, *_path) = path.split('/')
```

Comment:
It works for me when I use
```
_path = path.split('/')
sentinel = _path.pop(0)
```
Maybe you can think of something more elegant that still works in Python 2.7?